ql/src/java/org/apache/hadoop/hive/ql/parse/ExplainSemanticAnalyzer.java

```diff
@@ -156,7 +157,11 @@ public void analyzeInternal(ASTNode ast) throws SemanticException {
           while (driver.getResults(new ArrayList<String>())) {
           }
         } catch (CommandProcessorException e) {
-          throw new SemanticException(e.getMessage(), e);
+          if (e.getCause() instanceof ReCompileException) {
```

Comment:
I think right now we don't have any scenario in which this if would become true; this excepiont will only be thrown when it will be handled

Reply:
org.apache.hadoop.hive.cli.split8.TestMiniLlapLocalCliDriver[explainanalyze_2]